corehq/util/datadog/gauges.py

```diff
@@ -79,3 +80,29 @@ def _enforce_prefix(name, prefix):
         "Did you mean to call your gauge 'commcare.{}'? "
         "If you're sure you want to forgo the prefix, you can "
         "pass enforce_prefix=None".format(name))
+
+
+class datadog_track_errors(ContextDecorator):
+    """Record when something succeeds or errors in datadog
+
+    Eg: This code will log to commcare.myfunction.succeeded when it completes
+    successfully, and to commcare.myfunction.failed when an exception is
+    raised.
+
+        @datadog_track_errors('myfunction')
+        def myfunction():
+            pass
+    """
+
+    def __init__(self, name):
+        self.succeeded_name = "commcare.{}.succeeded".format(name)
+        self.failed_name = "commcare.{}.failed".format(name)
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if not exc_type:
+            datadog_counter(self.succeeded_name)
+        else:
+            datadog_counter(self.failed_name)
```

Comment:
I was thinking it might be helpful to include some representation of the exception as a tag, got any thoughts on how to do that?

Reply:
Maybe you could reuse the list from here: https://github.com/dimagi/commcare-hq/blob/00875a1e68ecce83168119cca015a6887fcce988/corehq/util/sentry.py#L18-L17

Reply:
Oh cool, yeah, that could work, I'll take a crack at it.